spark3/src/main/java/org/apache/iceberg/spark/actions/BaseSnapshotTableSparkAction.java

```diff
@@ -167,7 +167,7 @@ public SnapshotTable tableProperty(String property, String value) {
     // remove any possible location properties from origin properties
     properties.remove(LOCATION);
     properties.remove(TableProperties.WRITE_METADATA_LOCATION);
-    properties.remove(TableProperties.WRITE_FOLDER_STORAGE_LOCATION);
+    properties.remove(TableProperties.WRITE_DATA_LOCATION);
```

Comment:
When snapshotting a table that has  folder storage location set, should we also remove it in addition to write data location? 
Also should we remove object storage location?

Reply:
It'd be safer to remove both. Added in the new commit.

Reply:
I have a PR #2966 for this change, you can either also port my tests here, or remove this and I will update that PR once this is merged.

Reply:
Hi @jackye1995 , thanks for the information. I've added both `WRITE_FOLDER_STORAGE_LOCATION` and `OBJECT_STORE_PATH`. Can you rebase it in PR #2966 once this got merged?

Reply:
Sure, I can do that